pyigm/field/igmfield.py

```diff
@@ -0,0 +1,262 @@
+
+        Parameters
+        ----------
+        obj : Table
+          (can be anything that takes 'Z', and 'RA','DEC' in degrees)
+          Sources for calculation
+        comoving : bool, optional
+           If True then comoving, else physical
+        los_radec : SkyCoord, optional
+          Defaults to field RA/DEC
+
+        Returns
+        -------
+
+        rho : Quantity (array usually)
+          Impact parameter(s) in kpc
+        """
+        if los_coord is None:
+            los_coord = self.coord
+        # Coord
+        if isinstance(obj['RA'], Quantity):
+            ora = obj['RA']
+            odec = obj['DEC']
+        else:
+            ora = obj['RA']*u.deg
+            odec = obj['DEC']*u.deg
+
+        o_coord = SkyCoord(ra=ora, dec=odec)
+        ang_sep = o_coord.separation(los_coord).to('arcmin')
+        # Cosmology (kpc per arcmin)
+        if comoving:
+            kpc_amin = self.cosmo.kpc_comoving_per_arcmin(obj['Z'])
+        else:
+            kpc_amin = self.cosmo.kpc_proper_per_arcmin(obj['Z'])
+        rho = ang_sep * kpc_amin
+        # Return
+        return rho
+
+    def get_associated_galaxies(self, z, los_coord=None, R=300*u.kpc, dv_tol=500*u.km/u.s):
+        """Return a Table of associated galaxies for a given redshift and separation
+
+        Parameters
+        ----------
+        z : float
+          Redshift for association (usually IGM/CGM absorber redshift)
+        R : Quantity
+          Radius of impact parameter for association [300kpc]
+        dv_tol : Quantity
+          Velocity window for association [500km/s]
+        los_coord : SkyCoord, optional
+          Line-of-sight coordinates
+
+        Returns
+        -------
+        assoc_gal : Table
+          Table of associated galaxies (if any)
+        rho : Quantity
+          Impact parameters [kpc]
+        """
+        # los_radec
+        if los_coord is None:
+            los_coord = self.coord
+        # Cut on z first
+        dv_gal = const.c.to('km/s') * (self.galaxies['Z']-z)/(1+z)  # Approximate
+        gdz = np.where(np.abs(dv_gal)<dv_tol)[0]
+        if len(gdz) == 0:
+            return None
+        #
+        gdz_gal = self.galaxies[gdz]
+        rho = self.calc_rhoimpact(gdz_gal, los_coord)  # Could add this to Table
+        #
+        gd_rho = np.where(rho < R)[0]
+        if len(gd_rho) == 0:
+            return None
+        # Return
+        return gdz_gal[gd_rho], rho[gd_rho]
+
+    def get_observed(self, theta=None, subtab=None):
+        """Generate a Table of observed targets
+
+        Optionally to an angular distance from field center
+
+        Parameters
+        ----------
+        theta : Quantity or Angle, optional
+          Angular radius
+        subtab : Table, optional
+          User may input a table for processing
+          theta is ignored
+
+        Returns
+        -------
+        obs_targ : Table
+          Sub-table of targets that have been observed within this radius
```

Comment:
Sub-table of targets that have been observed in the field within `theta` (if given) and/or within `subtab` (if given).